### ISSUE TITLE
textbox & text alignment problems

### ISSUE THREAD

**Issue by shaehn** [chunyenHuang/hummusRecipe#122]
I have recently discovered while looking through the text tests that the centered text output no longer had its text drawn inside its textbox. I discovered that when the rotation angle is missing as an option, the code takes the wrong path causing the problem.

![image](https://user-images.githubusercontent.com/13596544/59453752-8c714380-8dde-11e9-97fb-d5d22b74acbd.png)

this is easily corrected by simply adding the default value for rotation to be zero in _getPathOptions.

Another problem appears when padding is involved. The text is pushed down out of the box.

![image](https://user-images.githubusercontent.com/13596544/59454498-38fff500-8de0-11e9-8443-e977938d80d6.png)

So the question I have here is should the textbox grow (font size takes precedence), or the font size shrink to keep the text inside the box (box size takes precedence)?

I have also noticed that the textbox is not taking into account descenders, so they appear outside the textbox.

![image](https://user-images.githubusercontent.com/13596544/59471430-12a27f80-8e09-11e9-9e13-cd3d51b6129b.png)

**Comment by shaehn** [comment]
John, I have taken the path of having the font size take precedence over the size of the text box, thus growing the size of the text box. Is this acceptable? Note, I have included the size of the descender characters. The following is what I came up with. No padding is being applied in the first example.

![image](https://user-images.githubusercontent.com/13596544/59482183-d6394880-8e35-11e9-9b3f-5b52b5f0fd80.png)

In the following, a padding of 5 has been applied.

![image](https://user-images.githubusercontent.com/13596544/59482319-5f507f80-8e36-11e9-8edd-6395eb603314.png)


If this is fine with you, I will generate a PR.

**Comment by atmortensen** [comment]
I'm having the same issue with the latest update. The text is being rendered above the textbox. 
![image](https://user-images.githubusercontent.com/24929893/59938438-328e0080-9412-11e9-84c9-f1638e585c88.png)

```
pdfRecipe.text('Test', 200, 500, {
     color: 'blue',
     fontSize: 12,
     textBox: {
          width: 100,
          height: 100,
          style: {
               stroke: 'blue'
          }
     }
});
```

Also the text box isn't even the right size. It should be a square in the example.

**Comment by shaehn** [comment]
Pull request #124 will fix the alignment problem. Note that assigning a height to the text box doesn’t do what you think it does. The text box effectively wraps the existing text. The width is truly the governing factor. The height automatically adjusts to the given text when it gets too long for the given width.

**Comment by shaehn** [comment]
Fix available in release 1.8.12

**Comment by Chrisjow** [comment]
> Pull request #124 will fix the alignment problem. Note that assigning a height to the text box doesn’t do what you think it does. The text box effectively wraps the existing text. The width is truly the governing factor. The height automatically adjusts to the given text when it gets too long for the given width.

Is it possible to disable this automatic behaviour in order to truncate the text instead of breaking lines?
